portfolio/src/main/webapp/comments.js

```diff
@@ -143,12 +157,53 @@ function toggleReplyField(commentId) {
 function addDeleteButton() {
   const button = document.createElement('button');
   button.innerText = 'Delete comments';
-  button.onclick = function () {
-    const request = new Request('/delete-data', {method: 'POST'});
+  button.onclick = function() {
+    const request = new Request('/delete-data', {method : 'POST'});
     fetch(request).then(() => {
-      window.location.reload(false); 
+      window.location.reload(false);
     });
   };
   button.classList.add('delete-comments-btn');
   document.getElementById('comments').appendChild(button);
 }
+
+// either adds log in button, or log out button. if user is
+// authenticated, displays user's email in DOM
+function addActionButtonsBasedOnAuthState() {
+  const authDiv = document.getElementById('auth');
+  if (authState.authenticated === true) {
+    const logOutLink = document.createElement('a');
+    const userEmailElement = document.createElement('p');
+    userEmailElement.innerText = 'Hello, ' + authState.email + ' !';
+    logOutLink.href = authState.logoutUrl;
```

Comment:
Suggestion: Prefer buttons for logout and login.

Reply:
I used `a` because server sends a single link, and I don't have any logic related to it yet

Reply:
Why do you want to use `button`?

Reply:
You can use button with a link :)